src/prefect/utilities/executors.py

```diff
@@ -260,19 +344,22 @@ def timeout_handler(
         "but continue running in the background."
     )
 
+    logger.debug(
+        f"Task '{name}': Falling back to daemonic soft limit timeout handler because "
+        f"we are running {soft_timeout_reason}."
+    )
     warnings.warn(msg, stacklevel=2)
     executor = ThreadPoolExecutor()
 
-    def run_with_ctx(*args: Any, _ctx_dict: dict, **kwargs: Any) -> Any:
-        with prefect.context(_ctx_dict):
-            return fn(*args, **kwargs)
+    def run_with_ctx(context: dict) -> Any:
+        with prefect.context(context):
+            return task.run(*args, **kwargs)
```

Comment:
I don't know why the args/kwargs were being passed down before since the `fn` (now `task.run`) was inherited from the context anyway. This seems like a good simplification to me.